smithy-typescript-ssdk-libs/server-node/src/index.spec.ts

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { mkdtemp } from "fs/promises";
+import { createServer, IncomingMessage, request, RequestOptions, Server, ServerResponse } from "http";
+import * as os from "os";
+import * as path from "path";
+import { Readable } from "stream";
+
+import { convertRequest } from "./node";
+
+let socketPath: string;
+let promiseResolve: ([req, res]: [IncomingMessage, ServerResponse]) => void;
+
+let server: Server;
+beforeAll(async () => {
+  server = createServer(function (req, res) {
+    promiseResolve([req, res]);
+    resToEnd = res;
+  });
+  // Create a temporary named pipe where to run the server and obtain a request
+  socketPath = path.join(await mkdtemp(path.join(os.tmpdir(), "named-pipe-for-test-")), "server");
+  // TODO Add support to Windows by using '\\\\?\\pipe'
+  // See: https://nodejs.org/api/net.html#identifying-paths-for-ipc-connections
+  server.listen(socketPath);
+});
+
+let resToEnd: ServerResponse;
```

Comment:
what does this variable name mean?

Reply:
What I tried to convey is that when the test ends, it should end this response. I guess I could use a better name and try what @everett1992 suggested on moving this logic inside the `getRequest` method.